src/shared/utilities/vsCodeUtils.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+export const localize = nls.loadMessageBundle()
+
+import { ErrorOrString, getLogger, Logger, LogLevel } from '../logger'
+
+// ------- Experimental combined output channel & traditional logger  --------------
+
+interface LogParams {
+    args: ErrorOrString[],
+    channel: vscode.OutputChannel,
+    level: LogLevel,
+    logger: Logger,
+    msg: string,
+    nlsKey: string,
+}
+
+// TODO: Leverage 3rd party code: https://raw.githubusercontent.com/Microsoft/vscode-nls/master/src/main.ts
+// function format(template: string, ...templateParams: any[]): string {
+// .......
+// }
+
+function log({ args = [], channel, level, msg, nlsKey, logger }: LogParams): void {
+    if (level === 'error') {
+        channel.show(true)
+    }
+    const [arg0, ...restArgs] = args
+
+    if (arg0 && arg0 instanceof Error) {
+        // TODO: Define standard/strategy. Sometimes err.message is part of msg. Probably shouldn't be.
+        channel.appendLine(
+            localize(nlsKey, msg, arg0.message, ...restArgs)
+        )
+        logger[level](
+            localize(nlsKey, msg, arg0.message, ...restArgs)
+            // TODO: Use english if/when we get multi lang support
```

Comment:
If we're logging a message because it was written to the output channel, I think it's fine to log the translated message (so that the log accurately represents what was written to the output channel).

Reply:
Sure, I can see this going either way or changing our minds. That's part of the reason for this wrapper. 

1. Reduce cut & paste verbosity
2. Enable consistence and ease of refactoring if we chance our minds